### PR TITLE
Add Jenkinsfile2 for Snakes-traffic-redirection pipeline

### PR DESCRIPTION
**Pull request by velychkovskyi**
## What Are We Doing Here?

We are creating new Jenkinsfile2 for second "Snakes-traffic-redirection" pipeline where deployment admin can manually change the weight of Blue/Green stacks and remove old Stack if needed. 

## How to Test and Verify

Follow the link: 
https://ci.devopsa3.me.uk/job/Snakes-traffic-redirection/
and while Running "Build with parameters" select the action
- Build without any action (checkout new version)
- Change weights of Blue/Green stacks
- Destroy Old Stack (all traffic will be redirected to new stack automatically) 

## Risk

Second press "Destroy Old Stack" will destroy leftover stack.

### Level

`Medium`

### Required Testing

`Manual`

### Risk Summary

Just initial commit for new functionality. 
Requires training before use in production.

## Code Review Checklist

- [ ] Testing instructions are included. If not, make sure you leave us a line or two for the reason.

- [ ] Well documented, including updates to any necessary markdown files. When
we inevitably come back to this code it will only take hours to figure out, not
days.

- [ ] Make sure PR has only the changes required for the Jira issues.

- [ ] Have you make sure to follow the agreements in our process document?

- [ ] Have someone from different region than you review your changes. If not, make sure to leave a comment on why it is not required

- [ ] If it's applicable, document deployment plan for the changes.

### CODE REVIEW

File: Jenkinsfile2 [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+pipeline {
+  agent {
+    label 'master'
+  }
+  options {
+    buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
+    timestamps()
+  }
+  parameters {
+    booleanParam(   name: 'UpdateALB', defaultValue: false, description: '')
```
Comment (mike-unitskyi) [inline Jenkinsfile2]: here and below and proper descriptions

File: Jenkinsfile2 [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    choice(         name: 'BlueWeight', choices: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], description: '')
+    choice(         name: 'GreenWeight', choices: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], description: '')
+    booleanParam(   name: 'RemovePreviousVersion', defaultValue: false, description: '')
+    string(         name: 'AWSRegion', defaultValue: 'us-east-1', description: 'Desired region for AWS resources')
+  }
+  environment {
+    OPSRepoURL      = 'git@github.com:IYermakov/DevOpsA3Training.git'
+    OPSRepoBranch   = 'weighted-tgs'
```
Comment (mike-unitskyi) [inline Jenkinsfile2]: why not master?

**Review by mike-unitskyi — CHANGES_REQUESTED** [review]

File: Jenkinsfile2 (hunk 1 above)
Comment (velychkovskyi) [reply Jenkinsfile2]: Added

File: Jenkinsfile2 (hunk 2 above)
Comment (velychkovskyi) [reply Jenkinsfile2]: master now

File: Jenkinsfile2 [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    label 'master'
+  }
+  options {
+    buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
+    timestamps()
+  }
+  parameters {
+    booleanParam(   name: 'UpdateALB', defaultValue: false, description: 'ALB TG weights will be updated unly if checkbox selected')
+    choice(         name: 'BlueWeight',  choices: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], description: 'Set the weight of ALB TG')
+    choice(         name: 'GreenWeight', choices: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], description: 'Set the weight of ALB TG')
```
Comment (mike-unitskyi) [inline Jenkinsfile2]: nitpick, does it make sense to set default values?
also, should `GreenWeight` be dependent on `BlueWeight`? like, `GreenWeight` + `BlueWeight` == 10?

Comment (vecinomio) [reply Jenkinsfile2]: There is no default value in choice parameter, but the  first item in the list is always default.

Comment (mike-unitskyi) [reply Jenkinsfile2]: so one of *Weigth parameters should be in DESC ordering, right?

**Comment by mike-unitskyi** [comment]
btw, jenkins check was failed, are you going to fix it?

File: Jenkinsfile2 (hunk 3 above)
Comment (vecinomio) [reply Jenkinsfile2]: Yes, you're right

File: Jenkinsfile2 [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+          catch (err) {
+            currentBuild.result = 'FAILURE'
+            throw (err)
+          }
+          echo "result is: ${currentBuild.currentResult}"
+        }
+      }
+    }
+  }
+}
```
Comment (mike-unitskyi) [inline Jenkinsfile2]: add new line

Comment (velychkovskyi) [reply Jenkinsfile2]: Done

**Review by mike-unitskyi — APPROVED** [review]

**Review by Emperorex — APPROVED** [review]
LGTM

**Pull request merged**